barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover.hpp

```diff
@@ -395,11 +393,22 @@ template <class ProverInstances_> class ProtoGalaxyProver_ {
 
         // Compute the combiner quotient polynomial as evaluations on points that are not in the vanishing set.
         //
+        constexpr FF inverse_two = FF(2).invert();
+        constexpr FF inverse_six = FF(6).invert();
         for (size_t point = ProverInstances::NUM; point < combiner.size(); point++) {
             auto idx = point - ProverInstances::NUM;
-            auto lagrange_0 = FF(1) - FF(point);
-            auto vanishing_polynomial = FF(point) * (FF(point) - 1);
-
+            FF lagrange_0;
+            FF vanishing_polynomial;
+            if constexpr (ProverInstances::NUM == 2) {
+                lagrange_0 = FF(1) - FF(point);
+                vanishing_polynomial = FF(point) * (FF(point) - 1);
+            } else if constexpr (ProverInstances::NUM == 3) {
+                lagrange_0 = (FF(1) - FF(point)) * (FF(2) - FF(point)) * inverse_two;
+                vanishing_polynomial = FF(point) * (FF(point) - 1) * (FF(point) - 2);
+            } else if constexpr (ProverInstances::NUM == 4) {
+                lagrange_0 = (FF(1) - FF(point)) * (FF(2) - FF(point)) * (FF(3) - FF(point)) * inverse_six;
+                vanishing_polynomial = FF(point) * (FF(point) - 1) * (FF(point) - 2) * (FF(point) - 3);
+            }
```

Comment:
maybe should add an else { ASSERT(false); } here to make sure nothing goes wrong if we ever add more k

Reply:
Yes, I think we should, but a static_assert